### PR TITLE
Adds support for Trino certificate authentication

### PR DESCRIPTION
**Pull request by austenLacy**
In the `0.310.0` release of `trino-python-client`, [support for Trino certificate authentication was added](https://github.com/trinodb/trino-python-client/pull/155). This PR bumps the `trino-python-client` and adds support for passing a client certificate and key through the connector.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,6 +103,7 @@ Please only use lower case schema names with this adapter.
 - [ldap](https://trino.io/docs/current/security/authentication-types.html) - Specify username in `user` and password in `password`
 - [kerberos](https://trino.io/docs/current/security/kerberos.html) - Specify username in `user`
 - [jwt](https://trino.io/docs/current/security/jwt.html) - Specify JWT token in `jwt_token`
+- [certificate](https://trino.io/docs/current/security/certificate.html) - Specify a client certificate in `client_certificate` and private key in `client_private_key`
```
Comment (findinpath) [inline README.md:106]: @hovaesco @Haunfelder  How does this change relate to https://github.com/starburstdata/dbt-trino/pull/24/files#r780898480 ?

**Review by findinpath — APPROVED** [review]

**Comment by findinpath** [comment]
@austenLacy  thank you for adding this improvement.

File: README.md (hunk 1 above)
Comment (hovaesco) [reply README.md:106]: `cert` parameter is used to specify client certificate path to connect with SSL/TLS.

`certificate` based authentication is used to validate certs provided by the Trino and Trino validates certs provided by a client.

These two don't interfere each other.

**Review by hovaesco — APPROVED** [review]

**Comment by austenLacy** [comment]
> @austenLacy see flake8 complaints, tests are failing with:
> 
> ```
> dbt/adapters/trino/connections.py:202:80: E501 line too long (82 > 79 characters)
> ```

Fixed

File: README.md (hunk 1 above)
Comment (hovaesco) [reply README.md:106]: I like the proposal, haven't seen a similar structure for other adapters, but it's worth considering. 

The reason behind `cert` name was to align with `trino-python-client` naming convention.

**Pull request merged**